raiden/tests/utils/transfer.py

```diff
@@ -63,12 +63,12 @@ def direct_transfer(initiator_app, target_app, asset, amount):
     initiator_app.raiden.api.transfer(
         asset,
         amount,
-        1,  # TODO: fill in identifier
-        target_app.raiden.address
+        target_app.raiden.address,
+        identifier,
```

Comment:
Yes it is valid since that also eventually passes via `transfer_async()`.

The reasoning I used in order to determine where to generate the identifier is to spot the one common function they all pass through and that was `transfer_async()` of transfer manager.